kernels/volk/volk_32f_stddev_and_mean_32f_x2.h

```diff
@@ -73,325 +73,513 @@
+  __m128 x0_reg, x1_reg;
+  __m128 f_reg;
+
+
+  for(;number < eigth_points; number++) {
+    v0_reg = _mm_load_ps(in_ptr);
+    in_ptr += 4;
+    __VOLK_PREFETCH(in_ptr + 4);
+
+    v1_reg = _mm_load_ps(in_ptr);
+    in_ptr += 4;    
+    __VOLK_PREFETCH(in_ptr + 4);
+
+    float n   = (float) number;
+    float np1 = n + 1.f;
+    f_reg = _mm_set_ps1(  1.f/( n*np1 ) );
+    
+    T0_acc = _mm_add_ps(T0_acc, v0_reg);
+
+    x0_reg = _mm_set_ps1(np1);
+    x0_reg = _mm_mul_ps(x0_reg, v0_reg);
+    x0_reg = _mm_sub_ps(x0_reg, T0_acc);
+    x0_reg = _mm_mul_ps(x0_reg, x0_reg);
+    x0_reg = _mm_mul_ps(x0_reg, f_reg);
+    S0_acc = _mm_add_ps(S0_acc, x0_reg);
+
+    T1_acc = _mm_add_ps(T1_acc, v1_reg);
+
+    x1_reg = _mm_set_ps1(np1);
+    x1_reg = _mm_mul_ps(x1_reg, v1_reg);
+    x1_reg = _mm_sub_ps(x1_reg, T1_acc);
+    x1_reg = _mm_mul_ps(x1_reg, x1_reg);
+    x1_reg = _mm_mul_ps(x1_reg, f_reg);
+    S1_acc = _mm_add_ps(S1_acc, x1_reg);
+  }
+
+  _mm_store_ps(&T[0], T0_acc);
+  _mm_store_ps(&T[4], T1_acc);
+  _mm_store_ps(&S[0], S0_acc);
+  _mm_store_ps(&S[4], S1_acc);
+
+  accrue_result( S, T, 8, eigth_points);
+
+  number = eigth_points*8;
+
+  FINALIZE:
+  for (; number < num_points; number++) {
+    update_square_sum_1_val(&S[0], &T[0], &number, in_ptr);
+    T[0] += (*in_ptr++);
   }
-  *stddev = returnValue;
-  *mean = newMean;
+
+  *stddev = sqrtf( S[0] / num_points );
+  *mean   = T[0] / num_points;
 }
-#endif /* LV_HAVE_GENERIC */
+#endif /* LV_HAVE_SSE */
 
+#ifdef LV_HAVE_AVX
+#include <immintrin.h>
 
+static inline void
+volk_32f_stddev_and_mean_32f_x2_a_avx(float* stddev, float* mean,
+                                         const float* inputBuffer,
+                                         unsigned int num_points)
+{
+  if (num_points == 0) { return; }
+
+  const float* in_ptr = inputBuffer;
+
+  unsigned int number = 1;
+
+  __VOLK_ATTR_ALIGNED(32) float T[16] = {0.f};
+  __VOLK_ATTR_ALIGNED(32) float S[16] = {0.f};
+
+  if (num_points < 16) {   
+    T[0] = (*in_ptr++);
+    goto FINALIZE; }
+
+  const unsigned int sixteenth_points = num_points / 16;
+
+  __m256 T0_acc = _mm256_load_ps(in_ptr);
+  in_ptr += 8;
+  __m256 T1_acc = _mm256_load_ps(in_ptr);
+  in_ptr += 8;
+
+  __m256 S0_acc = _mm256_setzero_ps();
+  __m256 S1_acc = _mm256_setzero_ps();
+  __m256 v0_reg, v1_reg;
+  __m256 x0_reg, x1_reg;
+  __m256 f_reg;
+
+  for(;number < sixteenth_points; number++) {
+    v0_reg = _mm256_load_ps(in_ptr);
+    in_ptr += 8;
+    __VOLK_PREFETCH(in_ptr + 8);
+
+    v1_reg = _mm256_load_ps(in_ptr);
+    in_ptr += 8;
+    __VOLK_PREFETCH(in_ptr + 8);
+
+    float n   = (float) number;
+    float np1 = n + 1.f;
+
+    f_reg = _mm256_set1_ps(  1.f/( n*np1 ) );
+    x0_reg = _mm256_set1_ps(np1);
+    
+    T0_acc = _mm256_add_ps(T0_acc, v0_reg);
+    
+    x0_reg = _mm256_mul_ps(x0_reg, v0_reg);
+    x0_reg = _mm256_sub_ps(x0_reg, T0_acc);
+    x0_reg = _mm256_mul_ps(x0_reg, x0_reg);
+    x0_reg = _mm256_mul_ps(x0_reg, f_reg);
```

Comment:
meta intrinsics for AVX would reduce copypasta from aligned to unaligned kernel. Same applies for SSE.

Reply:
I don't know how to write these, are there examples?

Reply:
[AVX intrinsics](https://github.com/gnuradio/volk/blob/master/include/volk/volk_avx_intrinsics.h) can be found here. I suggest you add your intrinsics in the appropriate files and just include them in the appropriate sections.

Reply:
This should be fine now